src/EndpointStandalone.sol

```diff
@@ -4,28 +4,54 @@ pragma solidity >=0.6.12 <0.9.0;
 import "./Endpoint.sol";
 import "./interfaces/IManagerStandalone.sol";
 import "./interfaces/IEndpointStandalone.sol";
+import "./libraries/Implementation.sol";
+import "./libraries/external/ReentrancyGuardUpgradeable.sol";
 
-abstract contract EndpointStandalone is IEndpointStandalone, Endpoint {
-    /// updating bridgeManager requires a new Endpoint deployment.
+abstract contract EndpointStandalone is
+    IEndpointStandalone,
+    Endpoint,
+    Implementation,
+    ReentrancyGuardUpgradeable
+{
+    bytes32 public constant ENDPOINT_MANAGER_SLOT =
+        bytes32(uint256(keccak256("endpoint.manager")) - 1);
+
+    /// @dev updating bridgeManager requires a new Endpoint deployment.
     /// Projects should implement their own governance to remove the old Endpoint contract address and then add the new one.
-    address immutable _manager;
+    function _getEndpointManagerStorage() internal pure returns (_Address storage $) {
+        uint256 slot = uint256(ENDPOINT_MANAGER_SLOT);
+        assembly ("memory-safe") {
+            $.slot := slot
+        }
+    }
 
     constructor(address manager) {
```

Comment:
This won't do anything behind a proxy. We probably want to set the manager in the initialiser and remove the constructor